mvt/ios/modules/mixed/viber.py

```diff
@@ -0,0 +1,154 @@
+        )
+        self.log.info("Found Viber database at path: %s", self.file_path)
+
+        self.conn = sqlite3.connect(self.file_path)
+        self.cur = self.conn.cursor()
+
+    def query_messages(self) -> None:
+        """Query records from the Viber messages table
+        joined with the phone number table.
+        """
+        self.cur.execute(
+            """
+            SELECT
+                ZVIBERMESSAGE.*,
+                ZPHONENUMBER.ZPHONE
+            FROM ZVIBERMESSAGE
+            LEFT JOIN ZPHONENUMBER ON
+                ZPHONENUMBER.Z_PK = ZVIBERMESSAGE.Z_PK;
+        """
+        )
+        # Get column names
+        self.col_names = [description[0] for description in self.cur.description]
+
+    def extract_messages(self) -> None:
+        """Extract messages and links from the query and add findings
+        to final results array.
+        """
+        # Loop through each message record
+        for message_row in self.cur:
+            # Create a dict for the current message containing all the keys & values
+            message = {}
+            for index, value in enumerate(message_row):
+                message[self.col_names[index]] = value
+
+            # Add an iso date key to the dict
+            message["isodate"] = convert_mactime_to_iso(message.get("ZDATE"))
+            # If record has no "ZTEXT" then create it with an empty string value
+            message["ZTEXT"] = message["ZTEXT"] if message["ZTEXT"] else ""
+
+            # Extract links from the Viber message.
+            message_links = []
+            fields_with_links = [
+                "ZTEXT",
+                "ZCLIENTMETADATA",
+            ]
+            for field in fields_with_links:
+                # If message contains a link, add it to the message_links list
+                if message.get(field):
+                    message_links.extend(check_for_links(message.get(field, "")))
+
+            # Filter out internal Viber URLs from the final filtered links list
+            self.filter_out_trusted_links(message_links)
+
+            # Add all the links found to the "list" key in the message dictionary
+            if self.filtered_links or (message.get("ZTEXT") or "").strip() == "":
+                message["links"] = list(set(self.filtered_links))
+            # Append current message to the results array
+            self.results.append(message)
+
+    def filter_out_trusted_links(self, message_links) -> None:
+        """Filter out Viber internal URLs from list of message links.
+        Operating under the assumption that links
+        from the official Viber domain are free of malware.
+        """
+        # TODO: Once we have more Viber message data verify that this wildcard combo
+        # is acceptable for filtering out internal URLs
+        self.filtered_links = []
+        for link in message_links:
+            if not (fnmatch.fnmatch(link, "*.viber.com/*")):
+                self.filtered_links.append(link)
```

Comment:
This could be updated when we have more information/data on what links we can filter out on the assumption that they are not going to be malicious